WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+        super.onSaveInstanceState(outState);
+
+        outState.putBoolean(KEY_IN_PROGRESS, mProgressDialog != null);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.menu_login, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.help) {
+            mLoginNavInputEmail.help();
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+     * autofill the username and password from BuildConfig/gradle.properties (developer feature,
+     * only enabled for DEBUG releases)
+     */
+    private void autofillFromBuildConfig() {
+        if (!BuildConfig.DEBUG) return;
+
+        String email = (String) WordPress.getBuildConfigValue(getActivity().getApplication(),
+                "DEBUG_DOTCOM_LOGIN_EMAIL");
+        if (!TextUtils.isEmpty(email)) {
+            mEmailEditText.setText(email);
+            AppLog.d(T.NUX, "Autofilled email from build config");
+        }
+    }
+
+    private void autocorrectEmail() {
+        if (mEmailAutoCorrected) {
+            return;
+        }
+        final String email = EditTextUtils.getText(mEmailEditText).trim();
+        // Check if the username looks like an email address
+        final Pattern emailRegExPattern = Patterns.EMAIL_ADDRESS;
+        Matcher matcher = emailRegExPattern.matcher(email);
+        if (!matcher.find()) {
+            return;
+        }
+        // It looks like an email address, then try to correct it
+        String suggest = EmailChecker.suggestDomainCorrection(email);
+        if (suggest.compareTo(email) != 0) {
+            mEmailAutoCorrected = true;
+            mEmailEditText.setText(suggest);
+            mEmailEditText.setSelection(suggest.length());
+        }
+    }
+
+    protected void onDoneAction() {
+        next(getEmail());
+    }
+
+    private boolean checkNetworkConnectivity() {
+        if (!NetworkUtils.isNetworkAvailable(getActivity())) {
+            FragmentTransaction ft = getFragmentManager().beginTransaction();
+            SignInDialogFragment nuxAlert;
+            nuxAlert = SignInDialogFragment.newInstance(getString(R.string.no_network_title),
+                    getString(R.string.no_network_message),
+                    R.drawable.ic_notice_white_64dp,
+                    getString(R.string.cancel));
+            ft.add(nuxAlert, "alert");
+            ft.commitAllowingStateLoss();
+            return false;
+        }
+        return true;
+    }
+
+    protected void next(String email) {
+        if (!checkNetworkConnectivity()) {
+            return;
+        }
+
+        if (isValidEmail(email)) {
+            showEmailCheckProgressDialog();
+            mDispatcher.dispatch(AccountActionBuilder.newIsAvailableEmailAction(email));
+        } else {
+            showEmailError(R.string.email_invalid);
+        }
+    }
+
+    private String getEmail() {
```

Comment:
I think this works. I'd probably call it `getTrimmedEmailInput` to reflect that it's coming from an input and that it's trimmed, but again this is just personal preference and I am totally fine with `getCleanedEmail`. It's much better than `getEmail` as that sounded like we are getting that from an AppPref or something like that.